### DISCUSSION TITLE
Add Tooltips for Custom Controls

### DISCUSSION THREAD

**Discussion by vavilon224** [fabricjs/fabric.js#9044] [Q&A] [answered]
I'm using `fabric v5.3.0`
I need to add a hover tooltip to icons, any ideas?
@ShaMan123 

```
  fabric.Object.prototype.setControlsVisibility({
    tl: true,
    mt: true,
    tr: true,
    ml: true,
    mr: true,
    bl: true,
    mb: true,
    br: true,
    mtr: false,
  });

  objectControls.tl = new fabric.Control({
    x: -0.5,
    y: -0.5,
    withConnection: true,
    actionName: 'remove',
    cursorStyle: 'pointer',
    render: renderIcon(returnImageFromUrl(iconTrashUrl), 'bgtrue', fabric),
    cornerSize: isMobile ? 30 : 50,
    mouseUpHandler: removeObject,
  });

```

**Comment by asturur** [comment, marked as answer]
You can sort of simulate an hover effect using the activeObject._corner property to verify if the current rendered control is the hovered one, but you can't really support tooltips out of the box.

You can get the control position on the canvas and in case position your tooltip there with html

**Reply by vavilon224** [reply]
`Got it thanks I'll try
`

**Reply by ShaMan123** [reply]
That is how I do it also
